templates/puma.rb

```diff
@@ -0,0 +1,20 @@
+# https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server
+
+# The environment variable WEB_CONCURRENCY may be set to a default value based
+# on dyno size. To manually configure this value use heroku config:set
+# WEB_CONCURRENCY.
+workers Integer(ENV['WEB_CONCURRENCY'] || 3)
+threads_count = Integer(ENV['MAX_THREADS'] || 5)
+threads(threads_count, threads_count)
+
+preload_app!
+
+rackup DefaultRackup
+port ENV['PORT'] || 3000
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.